map-structure/vi-map/src/vi-map.cc

```diff
@@ -3084,15 +3093,21 @@ bool VIMap::mergeAllSubmapsFromMapWithoutResources(
         // If this landmark has been remapped to a landmark in the base map we
         // can delete it or if it doesn't have any observations. Otherwise we
         // need to update the global landmark index of the base map.
+        const bool has_observations = vertex_copy->getLandmarks()
+                                          .getLandmark(owned_landmark_id)
+                                          .hasObservations();
         if (remapped_landmark_submap_ids.count(owned_landmark_id) > 0u ||
-            !vertex_copy->getLandmarks()
-                 .getLandmark(owned_landmark_id)
-                 .hasObservations()) {
+            !has_observations) {
           vertex_copy->getLandmarks().removeLandmark(owned_landmark_id);
-          LandmarkId invalid;
-          vertex_copy->updateIdInObservedLandmarkIdList(
-              owned_landmark_id, invalid);
-        } else {
+          if (!has_observations) {
+            LandmarkId invalid;
+            vertex_copy->updateIdInObservedLandmarkIdList(
+                owned_landmark_id, invalid);
+          }
+
+          // Check if the landmark is already included which can happen if the
+          // landmark was added as an observation from a previous vertex.
+        } else if (!landmark_index.hasLandmark(owned_landmark_id)) {
```

Comment:
what if it is included? essentially the `else` from the last `if`?

Reply:
In this case, the landmark observation was not remapped to an observation in the basemap and it is already included in the landmark index, so we don't have to do anything